text/0000-per-project-config.md

```diff
@@ -0,0 +1,67 @@
+- Feature Name: per-project-config
+- Start Date: 2019-05-01
+- RFC PR: (leave this empty)
+- Notion Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+Support Notion configuration of both hooks and platform information on a per-project basis using a `.notionrc` file in the project root.
+
+# Motivation
+[motivation]: #motivation
+
+We currently only support Notion configuration on a per-user basis in the `hooks.toml` file. This file allows users to change the URLs that tools are downloaded from. However, since a user could be working on both an open-source project (that uses the public registry) and an internal project (that uses a private registry), we instead want to allow the users to specify these settings on a per-project basis.
+
+Additionally, we currently already support per-project settings for the platform (node and yarn versions), specified in the `toolchain` key inside of `package.json`. Multiple users have requested an alternate method in a separate file, for various reasons (see https://github.com/notion-cli/notion/issues/282 for some discussion of one use-case). As we are already looking at adding per-project hooks to a separate file, that file should also support specifying the platform information.
+
+# Pedagogy
+[pedagogy]: #pedagogy
+
+We need to document the format of the `.notionrc` file, which will mostly match the format of `hooks.toml`, however it will also include the additional settings for `node` and `yarn` versions. Additionally, we need to document and explain the precedence order for when multiple configs are specified (e.g. user-level `hooks.toml` _and_ project-level `.notionrc`).
```

Comment:
> settings for `node` and `yarn` versions

We also support `npm` in `package.json`, if I'm not mistaken.

Reply:
We nominally support it, but it doesn't actually work. `npm` currently has to be set to the version that ships with `node`, and if it isn't then things will break. That said, you're right that we should include it in the discussion because the plan is for it to work eventually.